testsuite/testcases/src/state_sync_performance.rs

```diff
@@ -23,88 +19,98 @@ impl Test for StateSyncPerformance {
-            validator_synced_version
-        );
-        let start_instant = Instant::now();
-        let fullnode_metric_port = fullnode.expose_metric()?;
-        while fullnode
-            .counter(STATE_SYNC_COMMITTED_COUNTER_NAME, fullnode_metric_port)
-            .unwrap_or(0.0)
-            < validator_synced_version
-        {
-            thread::sleep(Duration::from_secs(1));
+        // Restart the fullnodes so they start syncing from a fresh state
+        for fullnode_id in &all_fullnodes {
+            let fullnode = ctx.swarm().full_node_mut(*fullnode_id).unwrap();
+            runtime.block_on(async { fullnode.start().await })?;
         }
-        println!(
-            "The fullnode has caught up to version: {}",
-            validator_synced_version
-        );
+
+        // Wait for all fullnodes to catch up to the highest synced version
+        info!("Restarting all the fullnodes and waiting for them to catchup.");
+        let timer = Instant::now();
+        runtime.block_on(async {
+            ctx.swarm()
+                .wait_for_all_nodes_to_catchup(Instant::now() + fullnode_sync_duration)
+                .await
+        })?;
+        let duration_to_state_sync = timer.elapsed();
+        let seconds_to_state_sync = duration_to_state_sync.as_secs();
 
         // Calculate the state sync throughput
-        let time_to_state_sync = start_instant.elapsed().as_secs();
-        if time_to_state_sync == 0 {
+        if seconds_to_state_sync == 0 {
             return Err(anyhow::format_err!(
                 "The time taken to state sync was 0 seconds! Something has gone wrong!"
             ));
         }
-        let state_sync_throughput = validator_synced_version as u64 / time_to_state_sync;
+        let state_sync_throughput = highest_synced_version as u64 / seconds_to_state_sync;
+
+        // Report the state sync results
         let state_sync_throughput_message =
-            format!("State sync throughput : {} txn/sec", state_sync_throughput,);
-        println!("Time to state sync: {:?} secs", time_to_state_sync);
-        // Display the state sync throughput and report the results
-        println!("{}", state_sync_throughput_message);
+            format!("State sync throughput : {} txn/sec", state_sync_throughput);
+        info!(
+            "Measured state sync throughput: {:?}",
+            state_sync_throughput_message
+        );
         ctx.report.report_text(state_sync_throughput_message);
         ctx.report.report_metric(
             self.name(),
             "state_sync_throughput",
             state_sync_throughput as f64,
         );
 
+        // Ensure we meet the success criteria.
+        // TODO: check_for_success is not really generic enough to check
+        // properties outside of txn_stat. This should be improved.
+        ctx.check_for_success(&txn_stat, &duration_to_state_sync)?;
```

Comment:
@JoshLind - May be I am missing something but where are we adding a success criteria for state sync throughput? `ctx.check_for_success` only checks for TPS/latency criteria IIUC.

Reply:
Good question. Is this not the success criteria? i.e., in local mode we assert 400 tps (https://github.com/aptos-labs/aptos-core/blob/0d1836e7044d0b55401208763b0ca3d0e6b90faa/testsuite/forge-cli/src/main.rs#L217) and in remote mode we assert 5k tps (https://github.com/aptos-labs/aptos-core/blob/0d1836e7044d0b55401208763b0ca3d0e6b90faa/testsuite/forge-cli/src/main.rs#L428).

Was there something else you wanted to check? Here I'm basically checking the state sync tps using `duration_to_state_sync`

Reply:
@JoshLind - That's the TPS success criteria during the txn emitter run. On top of that, you want to add a check `state_sync_throughput` above?

Reply:
Aah, indeed. But, if you look at the `check_for_success` function, it internally gets the number of txns committed during the run, divided by the duration you pass to it. Here, we're passing the number of txns committed (i.e., the size of the db), and the amount of time it took state sync to sync that state `duration_to_state_sync`. So, this is checking the state sync throughput, no? It's also why I added the comment -- it's not super clear. If you'd prefer, I can just check the throughput inline in the code and throw an error if it doesn't meet the expectation.

Reply:
Ah, I see. Indeed this is confusing. I would prefer if we check for `state_sync_throughput` inline here and error out if it doesn't meet the criteria. We can also add this as a separate success criteria if we need it for other tests but for now, inline check is okay I guess.

Reply:
Cool, thanks @sitalkedia. I've inlined the check for now. But, I agree. It would be nice if we could come up with a success criteria model that is more generic 😄